### PR TITLE
Stopasync improve

### PR DESCRIPTION
**Pull request by Kopilov**
Hello!

I have made some fixes for "StopRCallerAsync" method we added four months ago. (The main problem in production was that RtmpXXX directories became garbage, were not removed and wasted filesystem space.)

"TODO checking file length is wrong" is also fixed, with some refactoring.

### CODE REVIEW

**Comment by Kopilov** [comment]
If we move RCaller-3.0.1 from SNAPSHOT to release, I can publish my product (that is although just simple middleware, without specific R code) with RCaller-3.0.1 as a dependency :-)

**Comment by jbytecode** [comment]
@Kopilov tests seem to be passed. thank you for your contribution.

File: RCaller/src/test/java/com/github/rcaller/RunOnlineTest.java [hunk 1]
```diff
@@ -10,8 +10,8 @@
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertTrue;
+import static java.lang.System.currentTimeMillis;
+import static org.junit.Assert.*;
```
Comment (hola-holo) [inline RCaller/src/test/java/com/github/rcaller/RunOnlineTest.java:14]: our static analysis engine detects that your pull request includes a change from an explicit import to a wildcard import. According to the best practice (http://www.javadude.com/articles/importondemandisevil.html), import star (import on demand) is a bad practice, wildcard import can sometimes cause name conflicts and ambiguities.

Comment (Kopilov) [reply RCaller/src/test/java/com/github/rcaller/RunOnlineTest.java:14]: @hola-holo That was IDE autoreplacement, sorry!

Comment (Kopilov) [reply RCaller/src/test/java/com/github/rcaller/RunOnlineTest.java:14]: Fixed in https://github.com/jbytecode/rcaller/pull/38

**Comment by Kopilov** [comment]
> If we move RCaller-3.0.1 from SNAPSHOT to release, I can publish my product (that is although just simple middleware, without specific R code) with RCaller-3.0.1 as a dependency :-)

Published as https://github.com/Kopilov/RCallerService
The project is just copied from proprietary environment, no CI and tests are made at GitHub

**Pull request merged**